sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/ServiceBusReceivedMessage.java

```diff
@@ -3,63 +3,93 @@
 
 package com.azure.messaging.servicebus;
 
+import com.azure.core.amqp.models.AmqpAnnotatedMessage;
+import com.azure.core.amqp.models.AmqpBodyType;
+import com.azure.core.amqp.models.AmqpDataBody;
+import com.azure.core.amqp.models.BinaryData;
+import com.azure.core.util.logging.ClientLogger;
 import com.azure.messaging.servicebus.models.ReceiveMode;
 
 import java.time.Duration;
 import java.time.OffsetDateTime;
+import java.time.ZoneOffset;
 import java.util.Arrays;
-import java.util.HashMap;
+import java.util.Collections;
+import java.util.Date;
+import java.util.List;
 import java.util.Map;
-import java.util.Objects;
 import java.util.UUID;
+import java.util.stream.Collectors;
 
 /**
  * This class represents a received message from Service Bus.
  */
 public final class ServiceBusReceivedMessage {
+    private final ClientLogger logger = new ClientLogger(ServiceBusReceivedMessage.class);
+
+    private static final String ENQUEUED_TIME_UTC_NAME = "x-opt-enqueued-time";
```

Comment:
We already have a constants class.

Reply:
This is going to be really hard to manage random static strings copied in 5 places.